lib/elixir/lib/kernel/typespec.ex

```diff
@@ -613,6 +613,11 @@ defmodule Kernel.Typespec do
     {{kind, {name, arity}, spec}, caller.line}
   end
 
+  defp translate_spec(_kind, {_name, _meta, _args} = spec, _guard, caller) do
+    spec = Macro.to_string(spec)
+    compile_error caller, "function type specification missing return type: #{spec}"
```

Comment:
I feel like I should probably have noticed that given how much I've been working with specs lately... :P